open-sphere-plugins/feature-actions/src/main/java/io/opensphere/featureactions/editor/ui/FeatureActionExporter.java

```diff
@@ -0,0 +1,186 @@
+        myGroupList = groupList;
+    }
+
+    /**
+     * Launches the export feature action menu.
+     *
+     * @param parentDialog The parent dialog.
+     */
+    public void launch(Component parentDialog)
+    {
+        OptionDialog dialog = new OptionDialog(parentDialog);
+        dialog.setTitle("Export Feature Actions");
+        JPanel root = new JPanel();
+        StackLayout stack = new StackLayout();
+        stack.attach(root);
+        myName = new JTextField(myLayerName + " Feature Actions");
+        stack.add(new JLabel("Name:"));
+        stack.add(myName);
+        myButtonPanel = new RadioButtonPanel<>(New.list(EXPORT_ACTIVE, EXPORT_ALL), EXPORT_ACTIVE);
+        stack.add(myButtonPanel);
+        dialog.setComponent(root);
+        dialog.setModal(true);
+        dialog.buildAndShow();
+        if (dialog.getSelection() == JOptionPane.OK_OPTION)
+        {
+            myButtonPanel.getSelection();
+            try
+            {
+                Document doc = XMLUtilities.newDocument();
+                for (SimpleFeatureActionGroup featureActionGroup : myGroupList)
+                {
+                    for (SimpleFeatureAction featureAction : featureActionGroup.getActions())
+                    {
+                        if (checkExportValidity(featureAction))
+                        {
+                            Node baseNode = StateXML.createChildNode(doc, doc, doc, "/:featureActions", "featureActions");
+                            XMLUtilities.marshalJAXBObjectToElement(featureAction.getFeatureAction(), baseNode);
+                        }
+                    }
+                }
+                saveExportToFile(parentDialog, doc);
+            }
+            catch (ParserConfigurationException | XPathExpressionException | JAXBException e)
+            {
+                LOGGER.error(e.getMessage(), e);
+            }
+        }
+    }
+
+    /**
+     * Saves all the valid feature actions to the xml file.
+     *
+     * @param parentComponent The parent component.
+     * @param doc The document to save the feature actions to.
+     */
+    private void saveExportToFile(Component parentComponent, Document doc)
+    {
+        OutputStream outputStream;
+        MnemonicFileChooser chooser = new MnemonicFileChooser(myPrefsRegistry, getClass().getName());
+        chooser.setSelectedFile(new File(myName.getText() + ".xml"));
+        while (true)
+        {
+            int result = chooser.showSaveDialog(parentComponent, Collections.singleton(".xml"));
+            if (result == JFileChooser.APPROVE_OPTION)
+            {
+                File saveFile = chooser.getSelectedFile();
+                try
+                {
+                    outputStream = new FileOutputStream(saveFile);
+                    break;
+                }
+                catch (FileNotFoundException e)
+                {
+                    LOGGER.error("Failed to write to selected file [" + saveFile + "]: " + e, e);
```

Comment:
instead of relying on `saveFile.toString()` (which is implied), use `saveFile.getAbsolutePath()` instead, it gives a better indication of what the app was attempting to do.